### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Elhambasir**
### In this PR:

- Setup a Redux store
- Created a slice to save the state and reducers
- Created a directory that contains all Redux logic (/src/redux)
- Configured a Redux store (/src/redux/store.js)
- Defined a slice of state for books that:
  -   Includes an array of books (initial state: empty array)
  -   Includes a reducer that adds a book
  -   Includes a reducer that removes a book
- Defined a slice of state for categories that:
  -   Includes an array of categories (initial state: empty array)
  -   Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)

[See the project requirements.](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/add_reducers_an_actions_v2.md)

Thank you for reviewing my project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-scripts": "5.0.1",
+    "web-vitals": "^2.1.4"
```
Comment (bobb-Rob) [inline package.json]: - [x] Per the project requirements, `@reduxjs/toolkit` & `react-redux` should be present in package.json. Kindly install these two packages from `npm`.

File: src/redux/store.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { configureStore } from '@reduxjs/toolkit';
+
+const store = configureStore({
+  reducer: {},
+});
```
Comment (bobb-Rob) [inline src/redux/store.js]: - [x] In this store.js file, the following should have been implemented but you did not. Kindly ensure that you add the following.
   - * Imports `books` from `src/redux/books`.
   - * Imports `categories` from `src/redux/categories`.
   - * Uses `configureStore` to create a reducer.
  * `books` and `categories` should be added to `configureStore` method.

See example below 👇🏾 

```javascript
// ./src/redux/store.js

import { configureStore } from "@reduxjs/toolkit"
import booksReducer from "./books/booksSlice"
import categoriesReducer from "./categories/categoriesSlice"

const store = configureStore({
  reducer: { books: booksReducer, categories: categoriesReducer },
})

export default store
```

File: src/redux/categories/categoriesSlice.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const categorySlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+    checkStatus: () => 'Under construction',
+  },
+});
+export const { checkStatus } = categorySlice.actions;
+export default categorySlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/categories/categoriesSlice.js]: - [x] Please see below how your CategorySlice file code show be 👇🏾. Kindly implement the `checkStatus` reducer as shown below.

```javascript
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

File: src/redux/books/booksSlice.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    addBook: () => {
+      // code implementation to add a book
+    },
+    removeBook: () => {
+      // code implementation to remove a book
+    },
+  },
+});
+export const { addBook, removeBook } = bookSlice.actions;
+export default bookSlice.reducer;
```
Comment (bobb-Rob) [inline src/redux/books/booksSlice.js]: - [x] You did well with this booksSlice, but the `addBook` and `removeBook` reducers codes should be implemented. Kindly fix this.

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
+
```
Comment (bobb-Rob) [inline src/index.js:6]: - [x] [OPTIONAL] In your src directory, we expect to see a folder `component` which will contain all your project components.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Elhambasir, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (Elhambasir) [reply package.json]: Thank you I will do that.

File: src/redux/store.js (hunk 2 above)
Comment (Elhambasir) [reply src/redux/store.js]: Thank you for giving an example.

File: src/redux/categories/categoriesSlice.js (hunk 3 above)
Comment (Elhambasir) [reply src/redux/categories/categoriesSlice.js]: I got it. Thank you.

File: src/redux/books/booksSlice.js (hunk 4 above)
Comment (Elhambasir) [reply src/redux/books/booksSlice.js]: I thought that these should be implemented in the next step any way I will fix them.

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @Elhambasir , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- All requested changes implemented ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**